tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/type-test-expr.bal

```diff
@@ -1227,6 +1227,126 @@ function testBuiltInSubTypeTypeTestAgainstFiniteType() {
     assertFalse(b is Ints);
 }
 
+public type SomeRecord record {
+    int intField;
+};
+
+public function testIfElseWithTypeTest() {
+    SomeRecord? c = {intField: 10};
+    SomeRecord? f = ();
+    any x = true;
+
+    if c is () {
+
```

Comment:
Shouldn't the test fail if we reach here? If not, we're not testing the complete runtime behaviour right? Same applies to the other functions introduced.